scripts/test/modules/bins.py

```diff
@@ -0,0 +1,234 @@
+import os
+import threading
+import subprocess
+import fcntl
+
+import time
+
+from logger import log
+
+PLUGIN_DIR = "/etc/snap/plugins"
+SNAP_DIR = "/usr/local/bin"
+
+PLUGIN_URL = "http://snap.ci.snap-telemetry.io/plugin/build/latest/snap-plugin-collector-psutil"
+SNAP_URL = "http://snap.ci.snap-telemetry.io/snap/master/latest/snapd"
+SNAPCTL_URL = "http://snap.ci.snap-telemetry.io/snap/master/latest/snapctl"
+
+
+def _non_block_read(output):
+    fd = output.fileno()
+    fl = fcntl.fcntl(fd, fcntl.F_GETFL)
+    fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
+    try:
+        return output.readline()
+    except:
+        return ""
+
+
+class Binary(object):
+
+    def __init__(self, url, location):
+        self._url = url
+        self._dir = location
+        self._name = os.path.basename(url)
+
+    @property
+    def url(self):
+        return self._url
+
+    @url.setter
+    def url(self, u):
+        self._url = u
+
+    @property
+    def dir(self):
+        return self._dir
+
+    @dir.setter
+    def dir(self, loc):
+        self._dir = loc
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, n):
+        self._name = n
+
+    def __str__(self):
+        return self._name
+
+
+class Snapd(Binary, threading.Thread):
+
+    def __init__(self, url, location):
+        Binary.__init__(self, url, location)
+        threading.Thread.__init__(self)
```

Comment:
It leads to `RuntimeError: thread.__init__() not called`. I think to call `super` in this case would require to add `super` to each base class constructors as well ...